--- schema_old.rs	2025-04-14 18:43:16
+++ schema.rs	2025-04-14 18:43:03
@@ -65,6 +65,34 @@
 }
 
 diesel::table! {
+    component_balance (token_id, protocol_component_id, valid_to) {
+        token_id -> Int8,
+        new_balance -> Bytea,
+        previous_value -> Bytea,
+        balance_float -> Float8,
+        modify_tx -> Int8,
+        protocol_component_id -> Int8,
+        inserted_ts -> Timestamptz,
+        valid_from -> Timestamptz,
+        valid_to -> Timestamptz,
+    }
+}
+
+diesel::table! {
+    component_balance_default (token_id, protocol_component_id, valid_to) {
+        token_id -> Int8,
+        new_balance -> Bytea,
+        previous_value -> Bytea,
+        balance_float -> Float8,
+        modify_tx -> Int8,
+        protocol_component_id -> Int8,
+        inserted_ts -> Timestamptz,
+        valid_from -> Timestamptz,
+        valid_to -> Timestamptz,
+    }
+}
+
+diesel::table! {
     component_tvl (id) {
         id -> Int8,
         protocol_component_id -> Int8,
@@ -83,6 +111,36 @@
         modify_tx -> Int8,
         valid_from -> Timestamptz,
         valid_to -> Nullable<Timestamptz>,
+        inserted_ts -> Timestamptz,
+        modified_ts -> Timestamptz,
+    }
+}
+
+diesel::table! {
+    contract_storage (account_id, slot, valid_to) {
+        slot -> Bytea,
+        value -> Nullable<Bytea>,
+        previous_value -> Nullable<Bytea>,
+        account_id -> Int8,
+        modify_tx -> Int8,
+        ordinal -> Int8,
+        valid_from -> Timestamptz,
+        valid_to -> Timestamptz,
+        inserted_ts -> Timestamptz,
+        modified_ts -> Timestamptz,
+    }
+}
+
+diesel::table! {
+    contract_storage_default (account_id, slot, valid_to) {
+        slot -> Bytea,
+        value -> Nullable<Bytea>,
+        previous_value -> Nullable<Bytea>,
+        account_id -> Int8,
+        modify_tx -> Int8,
+        ordinal -> Int8,
+        valid_from -> Timestamptz,
+        valid_to -> Timestamptz,
         inserted_ts -> Timestamptz,
         modified_ts -> Timestamptz,
     }
@@ -141,6 +199,34 @@
 }
 
 diesel::table! {
+    protocol_state (protocol_component_id, attribute_name, valid_to) {
+        attribute_name -> Varchar,
+        attribute_value -> Bytea,
+        previous_value -> Nullable<Bytea>,
+        modify_tx -> Int8,
+        valid_from -> Timestamptz,
+        valid_to -> Timestamptz,
+        inserted_ts -> Timestamptz,
+        modified_ts -> Timestamptz,
+        protocol_component_id -> Int8,
+    }
+}
+
+diesel::table! {
+    protocol_state_default (protocol_component_id, attribute_name, valid_to) {
+        attribute_name -> Varchar,
+        attribute_value -> Bytea,
+        previous_value -> Nullable<Bytea>,
+        modify_tx -> Int8,
+        valid_from -> Timestamptz,
+        valid_to -> Timestamptz,
+        inserted_ts -> Timestamptz,
+        modified_ts -> Timestamptz,
+        protocol_component_id -> Int8,
+    }
+}
+
+diesel::table! {
     protocol_system (id) {
         id -> Int8,
         #[max_length = 255]
@@ -210,9 +296,19 @@
 diesel::joinable!(account_balance -> token (token_id));
 diesel::joinable!(account_balance -> transaction (modify_tx));
 diesel::joinable!(block -> chain (chain_id));
+diesel::joinable!(component_balance -> protocol_component (protocol_component_id));
+diesel::joinable!(component_balance -> token (token_id));
+diesel::joinable!(component_balance -> transaction (modify_tx));
+diesel::joinable!(component_balance_default -> protocol_component (protocol_component_id));
+diesel::joinable!(component_balance_default -> token (token_id));
+diesel::joinable!(component_balance_default -> transaction (modify_tx));
 diesel::joinable!(component_tvl -> protocol_component (protocol_component_id));
 diesel::joinable!(contract_code -> account (account_id));
 diesel::joinable!(contract_code -> transaction (modify_tx));
+diesel::joinable!(contract_storage -> account (account_id));
+diesel::joinable!(contract_storage -> transaction (modify_tx));
+diesel::joinable!(contract_storage_default -> account (account_id));
+diesel::joinable!(contract_storage_default -> transaction (modify_tx));
 diesel::joinable!(extraction_state -> block (block_id));
 diesel::joinable!(extraction_state -> chain (chain_id));
 diesel::joinable!(protocol_component -> chain (chain_id));
@@ -222,6 +318,10 @@
 diesel::joinable!(protocol_component_holds_contract -> protocol_component (protocol_component_id));
 diesel::joinable!(protocol_component_holds_token -> protocol_component (protocol_component_id));
 diesel::joinable!(protocol_component_holds_token -> token (token_id));
+diesel::joinable!(protocol_state -> protocol_component (protocol_component_id));
+diesel::joinable!(protocol_state -> transaction (modify_tx));
+diesel::joinable!(protocol_state_default -> protocol_component (protocol_component_id));
+diesel::joinable!(protocol_state_default -> transaction (modify_tx));
 diesel::joinable!(token -> account (account_id));
 diesel::joinable!(token_price -> token (token_id));
 diesel::joinable!(transaction -> block (block_id));
@@ -231,12 +331,18 @@
     account_balance,
     block,
     chain,
+    component_balance,
+    component_balance_default,
     component_tvl,
     contract_code,
+    contract_storage,
+    contract_storage_default,
     extraction_state,
     protocol_component,
     protocol_component_holds_contract,
     protocol_component_holds_token,
+    protocol_state,
+    protocol_state_default,
     protocol_system,
     protocol_type,
     token,
